### PR TITLE
Let HLS fallback to mpegts in case device reports unsupported container

### LINKED ISSUE #2214: Apple Lossless file not correctly streamed on iOS and macOS device
**Media Info of the file**

> Title Und ALAC stereo Default
Language und
Codec ALAC
Codec tag alac
Layout stereo
Channels 2 ch
Bitrate 1087 kbps
Sample rate 44100 Hz
Bit depth 16 bit
Default Yes

**Logs**

FFmpeg exited with code 1 after this:
```D:\Jellyfin\ffmpeg.exe -i file:"Z:\Music\AliA\realize\1-01 realize.m4a"  -map_metadata -1 -map_chapters -1 -threads 0   -acodec aac -strict experimental -ac 2 -vn -f hls -max_delay 5000000 -avoid_negative_ts disabled -start_at_zero -hls_time 3 -individual_header_trailer 0 -hls_segment_type aac -start_number 0 -hls_segment_filename "D:\JellyfinData\transcoding-temp\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\c762c3564cbe81faad88399689623d9c%d.aac" -hls_playlist_type vod -hls_list_size 0 -y "D:\JellyfinData\transcoding-temp\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\transcodes\c762c3564cbe81faad88399689623d9c.m3u8" ```

When manually invoke above command, ffmpeg is not happy with the `-hls_segment_type aac ` part. It seems like `aac` is not a valid type of `hls_segment_type`

**Stats for Nerds Screenshots**

Not available

**Server System (please complete the following information):**
 - OS: Windows Server 2019
 - Jellyfin Version: 10.4.3
 - Hardware settings & device: Quadro P400, Xeon D-1541(VM in ESXi, 16GB RAM)
 - Reverse proxy: N/A
 - Other hardware notes: Media mounted in CIFS/SMB share
 
**Client System (please complete the following information):**
 - Device: iPad Pro 2018
 - OS: iPadOS
 - Client: Browser
 - Browser (if Web client): Safari and Firefox
 - Client and Browser Version: Safari included in iPadOS 13.3, Firefox 2.10 App Store

**Extra note**
If I disable audio conversion those ALAC files actually plays in browser.  This is expected as iOS natives supports ALAC files, but the browser seems not exposing the capability correctly. Conversion works fine on Chrome on PC though, but `-hls_segment_type` became `mpegts` on Desktop. So basically, to make the situation perfect, there are 2 parts of this problem:
1. Adjust ffmpeg option as `-hls_segment_type aac` is invalid.
2. Prevent transcoding of alac files for iOS and macOS clients.

### PR DESCRIPTION
**Pull request by gnattu**
**Changes**

This PR will fallback the segment container to `mpegts` if the device reports a invalid container format. This fixes transcoded music playback on some devices (particularly Apple devices).  For more details please refer to the issue below.

**Issues**

Fixes #2214

### CODE REVIEW

File: MediaBrowser.Api/Playback/UniversalAudioService.cs [hunk 1]
```diff
@@ -300,6 +300,9 @@ private async Task<object> GetUniversalStream(GetUniversalAudioStream request, b
 
                 var transcodingProfile = deviceProfile.TranscodingProfiles[0];
 
+                //HLS Segment container can only be mpegts or fmp4 per ffmpeg documentation
+                var supoortedHLSContainer = new string[] { "mpegts", "fmp4" };
```
Comment (EraYaN) [inline MediaBrowser.Api/Playback/UniversalAudioService.cs]: typo and plural
```suggestion
                var supportedHLSContainers = new string[] { "mpegts", "fmp4" };
```

File: MediaBrowser.Api/Playback/UniversalAudioService.cs [hunk 2]
```diff
@@ -312,7 +315,8 @@ private async Task<object> GetUniversalStream(GetUniversalAudioStream request, b
                     PlaySessionId = playbackInfoResult.PlaySessionId,
                     StartTimeTicks = request.StartTimeTicks,
                     Static = isStatic,
-                    SegmentContainer = request.TranscodingContainer,
+                    //fallback to mpegts if device reports some wierd value that is not supported by HLS
+                    SegmentContainer = Array.Exists(supoortedHLSContainer, element => element == request.TranscodingContainer) ? request.TranscodingContainer : "mpegts",
```
Comment (EraYaN) [inline MediaBrowser.Api/Playback/UniversalAudioService.cs]:
```suggestion
                    SegmentContainer = Array.Exists(supportedHLSContainers, element => element == request.TranscodingContainer) ? request.TranscodingContainer : "mpegts",
```

File: MediaBrowser.Api/Playback/UniversalAudioService.cs [hunk 3]
```diff
@@ -312,7 +315,8 @@ private async Task<object> GetUniversalStream(GetUniversalAudioStream request, b
                     PlaySessionId = playbackInfoResult.PlaySessionId,
                     StartTimeTicks = request.StartTimeTicks,
                     Static = isStatic,
-                    SegmentContainer = request.TranscodingContainer,
+                    //fallback to mpegts if device reports some wierd value that is not supported by HLS
```
Comment (dkanada) [inline MediaBrowser.Api/Playback/UniversalAudioService.cs]:
```suggestion
                    // fallback to mpegts if device reports some weird value unsupported by hls
```

File: MediaBrowser.Api/Playback/UniversalAudioService.cs [hunk 4]
```diff
@@ -300,6 +300,9 @@ private async Task<object> GetUniversalStream(GetUniversalAudioStream request, b
 
                 var transcodingProfile = deviceProfile.TranscodingProfiles[0];
 
+                //HLS Segment container can only be mpegts or fmp4 per ffmpeg documentation
```
Comment (dkanada) [inline MediaBrowser.Api/Playback/UniversalAudioService.cs]:
```suggestion
                // hls segment container can only be mpegts or fmp4 per ffmpeg documentation
```

**Comment by JustAMan** [comment]
@gnattu could you please add some blurb basing on what I said above to the code, so we don't forget why this workaround was needed?

File: MediaBrowser.Api/Playback/UniversalAudioService.cs [hunk 5]
```diff
@@ -300,6 +300,9 @@ private async Task<object> GetUniversalStream(GetUniversalAudioStream request, b
 
                 var transcodingProfile = deviceProfile.TranscodingProfiles[0];
 
+                // hls segment container can only be mpegts or fmp4 per ffmpeg documentation
```
Comment (dkanada) [inline MediaBrowser.Api/Playback/UniversalAudioService.cs]:
```suggestion
                // hls segment container can only be mpegts or fmp4 per ffmpeg documentation
                // TODO: remove this when we switch back to the segment muxer
```

**Pull request merged**